ui/lib/core/addon/components/info-table-item-array.hbs

```diff
@@ -1,72 +1,68 @@
 {{! the class linkable-item is needed for the read-more component }}
-<div data-test-info-table-item-array {{did-insert (perform this.fetchOptions)}} class="linkable-item">
+<div data-test-info-table-item-array {{did-insert this.fetchOptions}} class="linkable-item">
   {{#if @isLink}}
     <div data-test-row-value={{@label}}>
-      <ReadMore>
-        {{#each this.displayArrayTruncated as |name|}}
-          {{#if (is-wildcard-string name)}}
-            {{#let (filter-wildcard name this.allOptions) as |wildcardCount|}}
-              <span>{{name}}</span>
-              <span class="tag is-light has-text-grey-dark" data-test-count="{{if wildcardCount wildcardCount 0}}">
-                includes
-                {{if wildcardCount wildcardCount 0}}
-                {{if (eq wildcardCount 1) @wildcardLabel (pluralize @wildcardLabel)}}
-              </span>
-              {{#if (eq this.displayArrayTruncated.lastObject name)}}
-                <LinkTo @route={{this.rootRoute}} @query={{hash tab=@queryParam}}>
-                  <span data-test-view-all={{lowercase @label}}>View all {{lowercase @label}}.</span>
-                </LinkTo>
-              {{/if}}
-            {{/let}}
-          {{else}}
-            {{#if (is-array this.itemRoute)}}
-              <LinkTo
-                @route={{(get this.itemRoute "0")}}
-                @models={{array (get this.itemRoute "1") name (get this.itemRoute "2")}}
-              >
-                {{name}}
-              </LinkTo>
+      {{#if this.fetchComplete}}
+        <ReadMore>
+          {{#each this.displayArrayTruncated as |item|}}
+            {{#if (is-wildcard-string item)}}
+              {{#let (filter-wildcard item this.allOptions) as |wildcardCount|}}
+                <span>{{item}}</span>
+                <span class="tag is-light has-text-grey-dark" data-test-count={{wildcardCount}}>
+                  {{if (not-eq wildcardCount undefined) (concat "includes " wildcardCount)}}
+                  {{if (eq wildcardCount 1) @wildcardLabel (pluralize @wildcardLabel)}}
+                </span>
+                {{#if (eq this.displayArrayTruncated.lastObject item)}}
+                  <LinkTo @route={{this.rootRoute}} @query={{hash tab=@queryParam}}>
+                    <span data-test-view-all={{lowercase @label}}>View all {{lowercase @label}}.</span>
+                  </LinkTo>
+                {{/if}}
+              {{/let}}
             {{else}}
-              <LinkTo
-                @route={{this.itemRoute}}
-                @model={{if @queryParam (concat @queryParam "/" name) name}}
-                data-test-item={{name}}
-              >
-                <span>{{name}}</span>
-              </LinkTo>
-            {{/if}}
-          {{/if}}
-          {{#if
```

Comment:
cleaned this up to just be `{{#if (not-eq item this.displayArrayTruncated.lastObject)}}` - am I thinking about that logic correctly?

Reply:
The logic got messy quickly, so I'd say if you tested an it worked, then we should be good.